src/adaptation/AccessCheck.cc

```diff
@@ -129,7 +128,7 @@ Adaptation::AccessCheck::checkCandidates()
     while (!candidates.empty()) {
         if (AccessRule *r = FindRule(topCandidate())) {
             /* BUG 2526: what to do when r->acl is empty?? */
-            acl_checklist = new ACLFilledChecklist(r->acl, filter.request);
+            const auto acl_checklist = new ACLFilledChecklist(r->acl, filter.request);
             acl_checklist->updateAle(filter.al);
             acl_checklist->updateReply(filter.reply);
             acl_checklist->syncAle(filter.request, nullptr);
```

Comment:
After this PR lands, we will post a PR that improves exception safety of the above and similar "create, configure, and nonBlockingCheck()" code sequences as promised [elsewhere](https://github.com/squid-cache/squid/pull/1827#discussion_r1622439751). Converting these essentially unused acl_checklist data members to local variables helps that future PR...

P.S. This comment does not request any PR code changes.